### PR TITLE
docs: add guides on deleting entities

### PR DESCRIPTION
**Pull request by yoonhyejin**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Demo
https://yoonhyejin.github.io/datahub-project-forked/docs/api/tutorials/deleting-entities-by-urn

resolves DR-352

### CODE REVIEW

File: docs/api/tutorials/deleting-entities-by-urn.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Deleting Entities By Urn
+
+## Why Would You Delete Entities?
```
Comment (jjoyce0510) [inline docs/api/tutorials/deleting-entities-by-urn.md:3]: We need a place where we introduce the notion of "Soft Deleted" vs "Hard Deleted"

Comment (jjoyce0510) [reply docs/api/tutorials/deleting-entities-by-urn.md:3]: (We assume people understand this, but most probably won't)

Comment (yoonhyejin) [reply docs/api/tutorials/deleting-entities-by-urn.md:3]: @jjoyce0510 Good, I'll move this to here! 
https://github.com/datahub-project/datahub/blob/9ea3a0baeb3b6306a9691dc6aa5759570a20e8ab/docs/api/tutorials/deleting-entities-by-urn.md?plain=1#L20

**Review by laulpogan — APPROVED** [review]
Looks good to me!

**Pull request merged**